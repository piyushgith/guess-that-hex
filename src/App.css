* {
  box-sizing: border-box;
}

body {
  margin: 0;
}

.box input, .input {
  font-size: 1.1em;
  opacity: 0.8;
  padding: 0.5em 1em;
  background-color: rgba(230, 230, 230, 0.4);
  border: none;
  border-bottom: 3px solid rgba(220, 220, 220, 0.4);
  outline: none;
}

.buttons {
  position: fixed;
  top: 0;
  right: 0;
  padding: 0.2em;
}

.btn {
  cursor: pointer;
  display: inline-block;
  padding: 0.5em 1em;
  background: rgba(230, 230, 230, 0.5);
  color: #555;
  border: none;
  font-weight: 100;
  font-size: 0.8em;
}

.color-container {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  pointer-events: none;
}
.color {
  width: 20%;
  height: 30vh;
  background: red;
  border-radius: 4px;
  display: flex;
  justify-content: center;
  align-items: center;
  font-weight: 600;
  font-size: 2em;
}


.GameControls, .GameStatus {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.GameControls {
  padding: 1em;
  background: rgba(255,255,255,0.8);
}

.GameStatus {
}

.GameStatus__waiting {

}

.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background-color: #222;
  height: 150px;
  padding: 20px;
  color: white;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
