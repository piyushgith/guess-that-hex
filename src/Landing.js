import React, { Component } from 'react';
import Radium from 'radium'
import Button from './shared/button'
import logo from './logo.png';
import globalStyle from './globalStyle'

class Landing extends Component {
  constructor(props) {
    super()
    this.state = {
      option: '',
      gameId: '',
      name: localStorage.getItem('gth:name') || ''
    }
    
  }

  componentDidMount() {
    let path = window.location.pathname.substr('1')
    console.log(path)
    if(path.length) {
      path = decodeURIComponent(path)
      console.log(path)
      this.setState({ gameId: path, option: 'join' })
    }
  }

  gameIdChanged = (e) => {
    this.setState({ gameId: e.target.value })
  }

  nameChanged = (e) => {
    const name = e.target.value
    this.setState({ name })
    localStorage.setItem('gth:name', name)
  }

  renderOptions = () => {
    return (
      <div style={styles.actions}>
        <Button
          id="new"
          key="new"
          style={{...styles.button, ...styles.actionButton}}
          onClick={() => this.setState({ option: 'new' })}
        >New Game</Button>
        <Button
          id="join"
          key="join"
          style={{...styles.button, ...styles.actionButton}}
          onClick={() => this.setState({ option: 'join' })}
        >Join Game</Button>
      </div>
    )
  }

  renderForm = () => {
    const { onCreate, onJoin } = this.props
    const { option, gameId, name } = this.state
    const buttonText = (option === 'new') ? 'Create Game' : 'Join Game'
    const submit = () => {
      option === 'new' ? onCreate(name) : onJoin(gameId, name)
    }

    return (
      <form style={styles.form} onSubmit={submit}>
        <div style={styles.formGroup}>
          <label style={[globalStyle.label, styles.label]} htmlFor="name">Your Name</label>
          <input style={[globalStyle.input, styles.input]} value={name} onChange={this.nameChanged} />
        </div>
        {option === 'join' && (
          <div style={styles.formGroup}>
            <label style={[globalStyle.label, styles.label]}  htmlFor="gameid">Game ID</label>
            <input style={[globalStyle.input, styles.input]} value={gameId} onChange={this.gameIdChanged} />
          </div>
        )}

        <Button type="submit" style={styles.submit}>{buttonText}</Button>
        <a
          role="button"
          style={styles.back}
          onClick={() => this.setState({ option: null })}
          key="back"
        >
          &laquo; Back
        </a>
      </form>
    )
  }

  render() {
    const { option } = this.state

    return (
      <div style={styles.page}>
        <div style={styles.container}>
          <div style={styles.header}>
            <img src={logo} style={styles.logo} alt="guess that hex" />
          </div>
          <div style={styles.content}>
            <h2 style={styles.heading}>Welcome to Guess that Hex!</h2>
            {option ? this.renderForm() : this.renderOptions()}
          </div>
        </div>
      </div>
    )
  }
}

const styles = {
  page: {
    backgroundColor: 'tomato',
    minHeight: '100vh',
    width: '100%',
    padding: '4em 0'
  },
  container: {
    backgroundColor: '#fafafa',
    borderRadius: '4px',
    overflow: 'hidden',
    width: "540px",
    maxWidth: "98%",
    margin: '4em auto'
  },
  header: {
    display: 'flex',
    justifyContent: 'center',
    background: '#eee',
    padding: '1.7em 0 0'
  },
  heading: {
    marginBottom: '2em'
  },
  form: {
    marginTop: '2em'
  },
  content: {
    padding: '2em 4em'
  },
  logo: {
    width: '300px',
    height: '92px',
    marginBottom: '-1.1em'
  },
  createContainer: {
    marginBottom: '2em'
  },
  formGroup: {
    marginBottom: '1.5em'
  },
  label: {
    display: 'block',
    fontWeight: '600',
    marginBottom: '0.2em'
  },
  input: {
    width: '100%',
  },
  back: {
    color: "#888",
    marginLeft: '1em',
    textDecoration: 'none',
    ':hover': {
      color: globalStyle.colors.primary,
      transition: 'all 200ms'
    }
  },
  submit: {
    marginTop: '1.5em'
  },
  actions: {
    display: 'flex',
    justifyContent: 'center',
    margin: '2em 0 1em'
  },
  actionButton: {
    margin: '0 0.5em'
  }
}

export default Radium(Landing);
