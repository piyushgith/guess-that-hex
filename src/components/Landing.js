import React, { Component } from 'react';
import PropTypes from 'prop-types'
import Button from '@material-ui/core/Button';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Grid from '@material-ui/core/Grid';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import MenuIcon from '@material-ui/icons/Menu';
import TextField from '@material-ui/core/TextField';
import Paper from '@material-ui/core/Paper';
import { withStyles } from '@material-ui/core/styles';
import logo from '../logo.png';

class Landing extends Component {
  static propTypes = {
    createGame: PropTypes.func.isRequired,
    joinGame: PropTypes.func.isRequired,
  }

  constructor(props) {
    super()
    this.state = {
      option: '',
      gameId: '',
      name: localStorage.getItem('gth:name') || ''
    }
  }

  componentDidMount() {
    const { option, gameId } = this.props
    this.setState({ option, gameId })
  }

  gameIdChanged = (e) => {
    this.setState({ gameId: e.target.value })
  }

  nameChanged = (e) => {
    const name = e.target.value
    this.setState({ name })
    localStorage.setItem('gth:name', name)
  }

  renderOptions = () => {
    const { classes } = this.props

    return (
      <div className={classes.actions}>
        <Button
          variant="contained"
          color="secondary"
          id="new"
          key="new"
          className={classes.actionButton}
          onClick={() => this.setState({ option: 'new' })}
        >New Game</Button>
        <Button
          variant="contained"
          color="secondary"
          id="join"
          key="join"
          onClick={() => this.setState({ option: 'join' })}
        >Join Game</Button>
      </div>
    )
  }

  renderForm = () => {
    const { classes, createGame, joinGame } = this.props
    const { option, gameId, name } = this.state
    const buttonText = (option === 'new') ? 'Create Game' : 'Join Game'
    const submit = (e) => {
      e.preventDefault()
      option === 'new' ? createGame(name) : joinGame(gameId, name)
    }

    return (
      <form className={classes.form} onSubmit={submit}>
        <TextField
          label="Your Name"
          className={classes.textField}
          margin="normal"
          value={name}
          helperText="Something to uniquely identify you"
          onChange={this.nameChanged}
          required
        />
        {option === 'join' && (
          <TextField
            label="Game ID"
            className={classes.textField}
            margin="normal"
            value={gameId}
            onChange={this.gameIdChanged}
            required
          />
        )}

        <div className={classes.formActions}>
          <Button
            type="submit"
            style={styles.submit}
            variant="contained"
            color="secondary"
          >
            {buttonText}
          </Button>
          <Button
            className={classes.back}
            onClick={() => this.setState({ option: null })}
            key="back"
          >
            &laquo; Back
          </Button>
        </div>
      </form>
    )
  }

  render() {
    const { option } = this.state
    const { classes } = this.props

    return (
      <div className={classes.page}>
        <Grid className={classes.grid} container justify="center">
          <Grid item xs={12} className={classes.row}>
            <Paper className={classes.pageContainer}>
              <Typography gutterBottom={true} align="center" variant="headline" component="h3">
                Welcome, friend
              </Typography>
              <Typography align="center" variant="body1" component="p">
                Let's get started guessing some hexes!
              </Typography>
              {option ? this.renderForm() : this.renderOptions()}
            </Paper>
          </Grid>
        </Grid>
      </div>
    )
  }
}

//<img src={logo} style={styles.logo} alt="guess that hex" />

const styles = theme => ({
  row: {
    maxWidth: '550px'
  },
  page: {
    display: 'flex',
    minHeight: '100vh',
    flexDirection: 'column'
  },
  pageContainer: {
    marginTop: theme.spacing.unit * 2,
    padding: '32px',
    paddingTop: '16px',
    [theme.breakpoints.up('md')]: {
      padding: '50px',
      paddingTop: '32px',
    },
  },
  textField: {
    width: '100%',
    maxWidth: '350px'
  },
  formActions: {
    marginTop: '1.5em',
    width: '100%',
    maxWidth: '310px'
  },
  grid: {
    backgroundColor: '#f5f5f5',
    flex: 1,
    padding: '12px'
  },
  form: {
    marginTop: theme.spacing.unit * 2,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center'
  },
  logo: {
    width: '300px',
    maxWidth: '90%',
    height: '92px',
    marginBottom: '-1.1em'
  },
  label: {
    display: 'block',
    fontWeight: '600',
    marginBottom: '0.2em'
  },
  input: {
    width: '100%',
  },
  back: {
    color: "#888",
    marginLeft: theme.spacing.unit,
  },
  submit: {
    marginTop: '1.5em'
  },
  actions: {
    display: 'flex',
    justifyContent: 'center',
    margin: '2em 0 1em'
  },
  actionButton: {
    margin: '0 0.5em'
  },
  root: {
    flexGrow: 1,
  }
})

export default withStyles(styles)(Landing)
