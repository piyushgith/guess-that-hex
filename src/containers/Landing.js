import React, { Component } from 'react';
import { connect } from 'react-redux'
import { createGame, joinGame } from '../reducers/game'
import LandingScreen from '../components/Landing'

class Landing extends Component {
  state = {
    option: null,
    gameId: null
  }

  componentWillMount() {
    const gameId = this.props.match.params.id
    const path = this.props.match.path
    let option = null
    if(path.startsWith('/join')) {
      option = 'join'
    } else if(path.startsWith('/new')) {
      option = 'new'
    }

    this.setState({ gameId, option })
  }

  render() {
    const { option, gameId } = this.state
    return (
      <LandingScreen 
        option={option}
        gameId={gameId}
        {...this.props}
      />
    )
  }
}

const mapStateToProps = state => ({})
const mapDispatchToProps = { createGame, joinGame }

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Landing)
