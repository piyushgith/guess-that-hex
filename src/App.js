import React, { Component } from 'react'
import { Provider } from 'react-redux'
import { Route, Switch } from 'react-router-dom'
import { ConnectedRouter } from 'connected-react-router'
import { MuiThemeProvider } from '@material-ui/core/styles';
import theme from './theme'
import { StyleRoot } from 'radium';
import store, { history } from './store'
import Landing from './containers/Landing'
import Game from './containers/Game'
import Page from './containers/Page'

const AppContainer = () => (
  <Provider store={store}>
    <MuiThemeProvider theme={theme}>
      <StyleRoot>
        <ConnectedRouter history={history}>
          <Route path="/" component={App} />
        </ConnectedRouter>
      </StyleRoot>
    </MuiThemeProvider>
  </Provider>
)

const App = () => (
  <Page>
    <Switch>
      <Route exact path="/" component={Landing} />
      <Route exact path="/join/:id" component={Landing} />
      <Route exact path="/new" component={Landing} />
      <Route path="/game/:id" component={Game} />
    </Switch>
  </Page>
)

export default AppContainer
