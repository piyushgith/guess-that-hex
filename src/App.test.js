import React from 'react'
import ReactDOM from 'react-dom'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import App from './App'
import Landing from './Landing'

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('renders the landing page', () =>  {
  const component = shallow(<App />)
  expect(component.find(Landing).length).toBe(1)
  expect(component.html()).toContain('Welcome to Guess that Hex')
  expect(toJson(component)).toMatchSnapshot()
})

test('renders a game as the creator', () => {
  jest.mock('./firebase/index')
  const component = mount(<App />)
  component.instance().createGame("Bill")
  console.log(component.html())
  expect(component.state('creator')).toBeTruthy()
  expect(component.html()).toContain('Other players can join by entering this game ID')
})

test('can join a game', () => {
  jest.mock('./firebase/index')
  const component = mount(<App />)
  component.instance().joinGame("abc", "Bill")
  console.log(component.html())
  expect(component.state('creator')).toBeFalsy()
  expect(component.html()).toContain("Waiting for game to start")
})

test('can create a new game', () =>  {
  const component = mount(<App />)
  component.find('#new').simulate('click')
  console.log(component.html())
  expect(toJson(component)).toMatchSnapshot()
})
