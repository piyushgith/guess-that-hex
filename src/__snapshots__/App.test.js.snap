// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can create a new game 1`] = `
<App>
  <div>
    <div
      style={Object {}}
    >
      <Landing
        onCreate={[Function]}
        onJoin={[Function]}
      >
        <div
          data-radium={true}
          style={
            Object {
              "backgroundColor": "tomato",
              "minHeight": "100vh",
              "padding": "4em 0",
              "width": "100%",
            }
          }
        >
          <div
            data-radium={true}
            style={
              Object {
                "backgroundColor": "#fafafa",
                "borderRadius": "4px",
                "margin": "4em auto",
                "maxWidth": "98%",
                "overflow": "hidden",
                "width": "480px",
              }
            }
          >
            <div
              data-radium={true}
              key=".0"
              style={
                Object {
                  "background": "#eee",
                  "display": "flex",
                  "justifyContent": "center",
                  "padding": "1.7em 0 0",
                }
              }
            >
              <img
                alt="guess that hex"
                data-radium={true}
                src="logo.png"
                style={
                  Object {
                    "height": "92px",
                    "marginBottom": "-1.1em",
                    "width": "300px",
                  }
                }
              />
            </div>
            <div
              data-radium={true}
              key=".1"
              style={
                Object {
                  "padding": "2em 3em",
                }
              }
            >
              <h2
                data-radium={true}
                key=".0"
                style={
                  Object {
                    "textAlign": "center",
                  }
                }
              >
                Welcome to Guess that Hex!
              </h2>
              <div
                data-radium={true}
                key=".1"
                style={
                  Object {
                    "display": "flex",
                    "justifyContent": "center",
                    "margin": "2em 0 1em",
                  }
                }
              >
                <Button
                  id="new"
                  key=".$new"
                  onClick={[Function]}
                  style={
                    Object {
                      "margin": "0 0.5em",
                    }
                  }
                >
                  <button
                    data-radium={true}
                    id="new"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#00aced",
                        "border": "none",
                        "borderRadius": "2px",
                        "color": "white",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "margin": "0 0.5em",
                        "padding": "0.9em 1em",
                      }
                    }
                  >
                    New Game
                  </button>
                </Button>
                <Button
                  id="join"
                  key=".$join"
                  onClick={[Function]}
                  style={
                    Object {
                      "margin": "0 0.5em",
                    }
                  }
                >
                  <button
                    data-radium={true}
                    id="join"
                    onClick={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#00aced",
                        "border": "none",
                        "borderRadius": "2px",
                        "color": "white",
                        "cursor": "pointer",
                        "fontSize": "16px",
                        "margin": "0 0.5em",
                        "padding": "0.9em 1em",
                      }
                    }
                  >
                    Join Game
                  </button>
                </Button>
              </div>
            </div>
          </div>
        </div>
      </Landing>
    </div>
  </div>
</App>
`;

exports[`renders the landing page 1`] = `
<div>
  <div
    style={Object {}}
  >
    <Landing
      onCreate={[Function]}
      onJoin={[Function]}
    />
  </div>
</div>
`;
